editor/block-mover/generate-order-title.js

```diff
@@ -0,0 +1,29 @@
+export default function( { typeTitle, position, isFirst, isLast, dir } ) {
+	if ( isFirst && isLast ) {
+		return `Block "${ typeTitle }" is the only block, and cannot be moved`;
+	}
+
+	if ( dir > 0 && ! isLast ) {
+		// moving down
+		return `Move "${ typeTitle }" block from position ${ position }` +
+			` down to position ${ position + 1 }`;
+	}
+
+	if ( dir > 0 && isLast ) {
+		// moving down, and is the last item
+		return `Block "${ typeTitle }" is at the end of the content and can’t be moved down`;
+	}
+
+	if ( dir < 0 && ! isFirst ) {
+		// moving up
+		return `Move "${ typeTitle }" block from position ${ position }` +
+			` up to position ${ position - 1 }`;
+	}
+
+	if ( dir < 0 && isFirst ) {
+		// moving up, and is the first item
+		return `Block "${ typeTitle }" is at the beginning of the content and can’t be moved up`;
+	}
+
+	return '';
```

Comment:
The subtle difference here in returning an empty string vs. returning nothing at all is that if it reaches this point and an empty string is returned, it will cause an empty [`aria-label` to be assigned](https://github.com/WordPress/gutenberg/blob/cb45cf800beedb7ff91410296468bdd4a760e2a0/components/icon-button/index.js#L17) to the button. We may want to remove this line altogether (an implicit `undefined` return), or `return null;` if we want to be explicit.